pkg/processor/build/runtime/runtime.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2017 The Nuclio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package runtime
+
+import (
+	"github.com/nuclio/nuclio/pkg/dockerclient"
+	"github.com/nuclio/nuclio/pkg/util/cmdrunner"
+
+	"github.com/nuclio/nuclio-sdk"
+	"github.com/pkg/errors"
+)
+
+type Runtime interface {
```

Comment:
Why did these become interfaces?

Reply:
The whole point was to separate common build stuff from the runtime-specific build stuff. An interface is warranted